### ISSUE TITLE
X11 / GUI Tools issue when using Exegol on WSL

### ISSUE THREAD

**Issue by TonyRode** [ShutdownRepo/Exegol#95]
Hey,

I freshly installed Exegol on an Ubuntu image under WSL, couldn't manage to get the GUI working.
The default build of the latest full image from an up to date Exegol repo was successful but without GUI support.

I got a full debug trace of the container creation process if needed.
Here's the short error message :
```
[?] Do you want to enable GUI? [y/N]: y
[!] Graphical interface sharing could not be enabled: Exegol tried to create a container with GUI support on a Windows host without having performed the availability tests
before.
```

Environment : 
  - Windows 11 (build 10.0.22000)
  - WSL 2 running Ubuntu (where Exegol has been installed)
  - And I meet the [requirements](https://tools.thehacker.recipes/exegol/getting-started/windows)
  
I previously did a
```
powershell> wsl --update
```

Docker Desktop is up and running and well configured (as on Shutdown's tutorial/requirements).

![Exegol_container_GUI_off](https://user-images.githubusercontent.com/4587307/171487189-77213e8d-24dc-4148-ba24-583c43d34dbc.png)

Any idea how we could fix it ?

**Comment by ShutdownRepo** [comment]
Hey there, thanks for raising this issue. 
Can you please run `exegol info -vvv` and make a screenshot (or copy-paste) of the output?

**Comment by TonyRode** [comment]
Yes sure, here it is : 

```
[*] Exegol is currently in version v4.0.2
[D] Pip installation: Off 🪓                                                                                                                                                 
[D] Git source installation: On ✔                                                                                                                                            
[D] Host OS: Windows                                                                                                                                                         
[D] Windows release: 10.0.22000                                                                                                                                              
[D] Python environment: WSL                                                                                                                                                  
[D] Docker engine: WSL2                                                                                                                                                      
[D] Docker desktop: On ✔                                                                                                                                                     
[D] Shell type: Linux                                                                                                                                                        

[V] Listing user configurations                                                                                                                                              

🧠 User configurations                                         
┌─────────────────────────────────────────────────────────────┐
│ Private workspace: /home/lama/.exegol/workspaces            │
│ Exegol resources: /home/lama/Exegol/Exegol/exegol-resources │
│ My resources: /home/lama/.exegol/my-resources               │
└─────────────────────────────────────────────────────────────┘

[V] Listing git repositories                                                                                                                                                 
[D] Loading git at /home/lama/Exegol/Exegol                                                                                                                                  
[D] Repo path: /home/lama/Exegol/Exegol/.git                                                                                                                                 
[D] Git repository successfully loaded                                                                                                                                       
[D] Git submodule repository detected                                                                                                                                        
[D] Loading git at /home/lama/Exegol/Exegol/exegol-docker-build                                                                                                              
[D] Repo path: /home/lama/Exegol/Exegol/.git/modules/sources                                                                                                                 
[D] Git repository successfully loaded                                                                                                                                       
[D] Git submodule repository detected                                                                                                                                        
[D] Loading git at /home/lama/Exegol/Exegol/exegol-resources                                                                                                                 
[D] Repo path: /home/lama/Exegol/Exegol/.git/modules/exegol-resources                                                                                                        
[D] Git repository successfully loaded                                                                                                                                       
[D] Fetch flags : 4                                                                                                                                                          
[D] Fetch note :                                                                                                                                                             
[D] Fetch old commit : None                                                                                                                                                  
[D] Fetch remote path : master                                                                                                                                               
[D] HEAD UP TO DATE flag detected                                                                                                                                            
[D] Fetch flags : 4                                                                                                                                                          
[D] Fetch note :                                                                                                                                                             
[D] Fetch old commit : None                                                                                                                                                  
[D] Fetch remote path : main                                                                                                                                                 
[D] HEAD UP TO DATE flag detected                                                                                                                                            
[D] Fetch flags : 4                                                                                                                                                          
[D] Fetch note :                                                                                                                                                             
[D] Fetch old commit : None                                                                                                                                                  
[D] Fetch remote path : main                                                                                                                                                 
[D] HEAD UP TO DATE flag detected                                                                                                                                            

🐙 Project modules                         
┌───────────┬────────────┬────────────────┐
│ Name      │ Status     │ Current branch │
├───────────┼────────────┼────────────────┤
│ Wrapper   │ Up to date │ master         │
│ Images    │ Up to date │ main           │
│ Resources │ Up to date │ main           │
└───────────┴────────────┴────────────────┘

[D] Fetching remote image tags, digests and sizes                                                                                                                            
[D] Fetching information from: https://hub.docker.com/v2/repositories/nwodtuhs/exegol/tags                                                                                   
[D] └── nightly → (remote) sha256:f32095a44b8702b9030aba26750e89aace61216cec8662c35f3bc921245ad904                                                                           
[D] └── full-2.0.2      → (remote) sha256:7d4a12cc5a1e9844f133c8888ed35c3aaceceed1911fb7aace946a8ffe77a232                                                                   
[D] └── full    → (remote) sha256:7d4a12cc5a1e9844f133c8888ed35c3aaceceed1911fb7aace946a8ffe77a232                                                                           
[D] └── light-2.0.2     → (remote) sha256:630b085d3d8f4e077844ea9b530c4a5856d2db9f7abdcf5ece2ae87fadc80d4d                                                                   
[D] └── light   → (remote) sha256:630b085d3d8f4e077844ea9b530c4a5856d2db9f7abdcf5ece2ae87fadc80d4d                                                                           
[D] └── ad-2.0.2        → (remote) sha256:a141d1c4e4f50d481a0ab8f2a4d60161c4a099f388485e6f1f8042ed66065611                                                                   
[D] └── ad      → (remote) sha256:a141d1c4e4f50d481a0ab8f2a4d60161c4a099f388485e6f1f8042ed66065611                                                                           
[D] └── web-2.0.2       → (remote) sha256:f1a210d22838306b14fbd0afcfda51103464239cfb75b752f19bf1b3a8e098be                                                                   
[D] └── web     → (remote) sha256:f1a210d22838306b14fbd0afcfda51103464239cfb75b752f19bf1b3a8e098be                                                                           
[D] └── osint-2.0.2     → (remote) sha256:d63405881d637711ac2f0a09a321e84ad9caae743cb3b0acc7e9f6ff2558a668                                                                   
[D] Fetching information from: https://hub.docker.com/v2/repositories/nwodtuhs/exegol/tags?page=2                                                                            
[D] └── osint   → (remote) sha256:d63405881d637711ac2f0a09a321e84ad9caae743cb3b0acc7e9f6ff2558a668                                                                           
[D] └── full-2.0.1      → (remote) sha256:8ff1870bd0dc5c7d79772454b67651d79a23f2b6ad15ab490f00822f6ee324f1                                                                   
[D] └── ad-2.0.1        → (remote) sha256:55e5271e6d8a73d745c3200ce4cb0298b4063e20dfd9f762fc506ca359805d51                                                                   
[D] └── web-2.0.1       → (remote) sha256:ca9c34fb433797ee51199e75b0cebc051fde9b1fe837a03ca5b286478cc05250                                                                   
[D] └── light-2.0.1     → (remote) sha256:9e7f1228f18ec1cc687d2e45fe6ca50013dc30a2f2b0ff4a9cd2bfcbb468dc5d                                                                   
[D] └── osint-2.0.1     → (remote) sha256:7ea562eb869ccc1b729e6468af151b0e4f3d1aeb739baf5d9f20207c5f8b104a                                                                   
[D] └── web-2.0.0       → (remote) sha256:f74fcc92b8e3f31f45e4141fb9eb7f9e85c29e61785eda85b2aa989c56d94571                                                                   
[D] └── light-2.0.0     → (remote) sha256:18fc89446474dfd8ceb2951df41baa571447e4eb6ea7df1b127d8cd6caea222c                                                                   
[D] └── osint-2.0.0     → (remote) sha256:e5e1832f98ce7cc69b9317bab5a9374e2a9d3c0489c36db6572c6ca4387d86b1                                                                   
[D] └── full-2.0.0b5    → (remote) sha256:252afb0667c3d51670d53fdfb53550d71fc7a4d45c4f326b158ff346f0bb0b14                                                                   
[D] Fetching information from: https://hub.docker.com/v2/repositories/nwodtuhs/exegol/tags?page=3                                                                            
[D] └── ad-2.0.0b5      → (remote) sha256:42ac024f403e51dbd5d500afad3750b7666456fa22348740412071ec90481108                                                                   
[D] └── web-2.0.0b5     → (remote) sha256:cc4966d12043bc6a8d98647ab185eef1d157b0bc9103d5ec2cebb3b6e91b98c1                                                                   
[D] └── light-2.0.0b5   → (remote) sha256:b0409163fc61ac9e987843894c9ac53bf117bedce3f36f019115c668fc0c5123                                                                   
[D] └── osint-2.0.0b5   → (remote) sha256:2f693307e10d40785ec0724ccc8b28ef7b7ca4ff6efd21c66da71d0af7d432cd                                                                   
[D] Fetching local image tags, digests (and other attributes)                                                                                                                
[D] == Loading container : exegol-full_2                                                                                                                                     
[D] Parsing envs : PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                         
[D] Adding env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                             
[D] Capabilities : []                                                                                                                                                        
[D] Load devices : []                                                                                                                                                        
[D] Parsing mount : {'Type': 'bind', 'Source': '/etc/localtime', 'Destination': '/etc/localtime', 'Mode': '', 'RW': False, 'Propagation': 'rprivate'}                        
[D] Parsing mount : {'Type': 'bind', 'Source': '/etc/timezone', 'Destination': '/etc/timezone', 'Mode': '', 'RW': False, 'Propagation': 'rprivate'}                          
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/.exegol/my-resources', 'Destination': '/my-resources', 'Mode': '', 'RW': True, 'Propagation': 'rprivate'}         
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/Exegol/Exegol/exegol-resources', 'Destination': '/opt/resources', 'Mode': '', 'RW': True, 'Propagation':          
'rprivate'}                                                                                                                                                                  
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/.exegol/workspaces/full_2', 'Destination': '/workspace', 'Mode': '', 'RW': True, 'Propagation': 'rprivate'}       
[D] Loading workspace volume source : /home/lama/.exegol/workspaces/full_2                                                                                                   
[D] Private workspace detected                                                                                                                                               
[D] └── full    → (remote) sha256:7d4a12cc5a1e9844f133c8888ed35c3aaceceed1911fb7aace946a8ffe77a232                                                                           
[D] == Loading container : exegol-training                                                                                                                                   
[D] Parsing envs : PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                         
[D] Adding env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                             
[D] Capabilities : []                                                                                                                                                        
[D] Load devices : []                                                                                                                                                        
[D] Parsing mount : {'Type': 'bind', 'Source': '/etc/localtime', 'Destination': '/etc/localtime', 'Mode': '', 'RW': False, 'Propagation': 'rprivate'}                        
[D] Parsing mount : {'Type': 'bind', 'Source': '/etc/timezone', 'Destination': '/etc/timezone', 'Mode': '', 'RW': False, 'Propagation': 'rprivate'}                          
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/.exegol/my-resources', 'Destination': '/my-resources', 'Mode': '', 'RW': True, 'Propagation': 'rprivate'}         
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/Exegol/Exegol/exegol-resources', 'Destination': '/opt/resources', 'Mode': '', 'RW': True, 'Propagation':          
'rprivate'}                                                                                                                                                                  
[D] Parsing mount : {'Type': 'bind', 'Source': '/home/lama/Exegol/Exegol', 'Destination': '/workspace', 'Mode': '', 'RW': True, 'Propagation': 'rprivate'}                   
[D] Loading workspace volume source : /home/lama/Exegol/Exegol                                                                                                               
[D] Custom workspace detected                                                                                                                                                
[D] └── full    → (remote) sha256:7d4a12cc5a1e9844f133c8888ed35c3aaceceed1911fb7aace946a8ffe77a232                                                                           
[V] Listing local and remote Exegol images                                                                                                                                   

🛸 Available images                                                                                   
┌───────────────┬───────────┬───────────────┬──────────────┬──────────────────┬──────────────────────┐
│ Id            │ Image tag │ Download size │ Size on disk │ Build date (UTC) │ Status               │
├───────────────┼───────────┼───────────────┼──────────────┼──────────────────┼──────────────────────┤
│ 9d8554027ccb  │ full      │ 12.6GB        │ 27.3GB       │ 20/05/2022 16:00 │ Up to date (v.2.0.2) │
│ Not installed │ nightly   │ 12.0GB        │ N/A          │ N/A              │ Not installed        │
│ Not installed │ light     │ 4.3GB         │ N/A          │ N/A              │ Not installed        │
│ Not installed │ ad        │ 9.3GB         │ N/A          │ N/A              │ Not installed        │
│ Not installed │ web       │ 7.3GB         │ N/A          │ N/A              │ Not installed        │
│ Not installed │ osint     │ 3.1GB         │ N/A          │ N/A              │ Not installed        │
└───────────────┴───────────┴───────────────┴──────────────┴──────────────────┴──────────────────────┘

[V] Listing local Exegol containers                                                                                                                                          
[*] Number of Exegol containers: 2

👽 Available containers                                                                                                                                                      
┌────────────┬───────────────┬─────────┬───────────┬─────────────────────────┬─────────────────────────────────────┬─────────┬───────┬──────────────────────────────────────┐
│ Id         │ Container tag │ State   │ Image tag │ Configurations          │ Mounts                              │ Devices │ Ports │ Envs                                 │
├────────────┼───────────────┼─────────┼───────────┼─────────────────────────┼─────────────────────────────────────┼─────────┼───────┼──────────────────────────────────────┤
│ 40366a72d2 │ full_2        │ Stopped │ full      │ Privileged: Off ✔       │ /etc/localtime ➡ /etc/localtime     │         │       │ PATH=/usr/local/sbin:/usr/local/bin… │
│            │               │         │           │ GUI: Off 🪓             │ (RO)                                │         │       │                                      │
│            │               │         │           │ Network mode: bridge    │ /etc/timezone ➡ /etc/timezone (RO)  │         │       │                                      │
│            │               │         │           │ Share timezone: On ✔    │ /home/lama/.exegol/my-resources ➡   │         │       │                                      │
│            │               │         │           │ Exegol resources: On ✔  │ /my-resources                       │         │       │                                      │
│            │               │         │           │ My resources: On ✔      │ /home/lama/Exegol/Exegol/exegol-re… │         │       │                                      │
│            │               │         │           │                         │ ➡ /opt/resources                    │         │       │                                      │
│            │               │         │           │                         │ /home/lama/.exegol/workspaces/full… │         │       │                                      │
│            │               │         │           │                         │ ➡ /workspace                        │         │       │                                      │
│            │               │         │           │                         │                                     │         │       │                                      │
│ c2ab88cfcd │ training      │ Stopped │ full      │ Privileged: Off ✔       │ /etc/localtime ➡ /etc/localtime     │         │       │ PATH=/usr/local/sbin:/usr/local/bin… │
│            │               │         │           │ GUI: Off 🪓             │ (RO)                                │         │       │                                      │
│            │               │         │           │ Network mode: bridge    │ /etc/timezone ➡ /etc/timezone (RO)  │         │       │                                      │
│            │               │         │           │ Share timezone: On ✔    │ /home/lama/.exegol/my-resources ➡   │         │       │                                      │
│            │               │         │           │ Exegol resources: On ✔  │ /my-resources                       │         │       │                                      │
│            │               │         │           │ My resources: On ✔      │ /home/lama/Exegol/Exegol/exegol-re… │         │       │                                      │
│            │               │         │           │                         │ ➡ /opt/resources                    │         │       │                                      │
│            │               │         │           │                         │ /home/lama/Exegol/Exegol ➡          │         │       │                                      │
│            │               │         │           │                         │ /workspace                          │         │       │                                      │
│            │               │         │           │                         │                                     │         │       │                                      │
└────────────┴───────────────┴─────────┴───────────┴─────────────────────────┴─────────────────────────────────────┴─────────┴───────┴──────────────────────────────────────┘
```

I only tried to create a new container (full_2) to make tests since my previous message.

**Comment by ShutdownRepo** [comment]
Awesome, thank you for this, it will help us better understand your env.
Can you try to run `exegol update -v` and then type `dev` when asked to select from which branch the wrapper code needs to be updated? We never know, maybe the issue you are facing was fixed in between release 4.0.2 and now (4.0.3b1)

**Comment by TonyRode** [comment]
Thanks for helping :)

I did the update from dev as you mentionned, and tried a new container creation. Same result.

![Exegol_container_GUI_off_v4 0 3b1](https://user-images.githubusercontent.com/4587307/171508691-e1a5e373-9cff-4557-ac5d-565d037d8ea0.png)

**Comment by Dramelac** [comment]
Hello

Thank you for raising this issue ! I think the error is raised by the interactive function, you can try in CLI mode with the following command:
`exegol start test full`

Let me know if this bypass worked for you.

I will fix this error in the next 4.0.3 release.

Thank you

**Comment by TonyRode** [comment]
Hey Dramelac

Unfortunately it didn't work. Same error.
I did `exegol update`, everything was already up to date (dev and main branches), then `exegol start test full`.

```
lama@MSI:/mnt/c/WINDOWS/system32$ exegol start test2 full
[*] Exegol is currently in version v4.0.3b1
[*] Starting exegol
[*] Arguments supplied with the command, skipping interactive mode
[!] Graphical interface sharing could not be enabled: Exegol tried to create a container with GUI support on a Windows host without having performed the availability tests
before.
[!] Docker desktop (Windows & macOS) does not support sharing of host network interfaces.
[*] To share network ports between the host and exegol, use the --port parameter.

⭐ Container summary
┌──────────────────┬────────────────────────┐
│             Name │ test2                  │
│            Image │ full - v.2.0.2         │
├──────────────────┼────────────────────────┤
│              GUI │ Off 🪓                 │
│          Network │ bridge                 │
│         Timezone │ On ✔                   │
│ Exegol resources │ On ✔ (/opt/resources)  │
│     My resources │ On ✔ (/my-resources)   │
│              VPN │ N/A                    │
│       Privileged │ Off ✔                  │
│        Workspace │ Dedicated (/workspace) │
└──────────────────┴────────────────────────┘

[*] Creating new exegol container
[+] Exegol container successfully created !
[*] Location of the exegol workspace on the host : /mnt/c/home/lama/.exegol/workspaces/test2
[+] Opening shell in Exegol 'test2'
```

**Comment by Dramelac** [comment]
Fixed in dev with commit 99ac8d5d99be8899eecf251cff72f28f9163b32a 
Will be release in 4.0.3.
